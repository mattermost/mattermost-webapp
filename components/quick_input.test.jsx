// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {mount} from 'enzyme';

import QuickInput from 'components/quick_input';

describe('components/QuickInput', () => {
    describe('should not render clear button', () => {
        [
            ['in default state', {}],
            ['when not clearable', {value: 'value', clearable: false, onClear: () => {}}],
            ['when no onClear callback', {value: 'value', clearable: true}],
            ['when value undefined', {clearable: true, onClear: () => {}}],
            ['when value empty', {value: '', clearable: true, onClear: () => {}}],
        ].forEach(([description, props]) => {
            it(description, () => {
                const wrapper = mount(
                    <QuickInput {...props}/>,
                );

                expect(wrapper.find('.input-clear').exists()).toBe(false);
            });
        });
    });

    describe('should render clear button', () => {
        test('with default tooltip text', () => {
            const wrapper = mount(
                <QuickInput
                    value='mock'
                    clearable={true}
                    onClear={() => {}}
                />,
            );

            expect(wrapper.find('.input-clear')).toMatchSnapshot();
        });

        test('with customized tooltip text', () => {
            const wrapper = mount(
                <QuickInput
                    value='mock'
                    clearable={true}
                    clearableTooltipText='Custom'
                    onClear={() => {}}
                />,
            );

            expect(wrapper.find('.input-clear')).toMatchSnapshot();
        });

        test('with customized tooltip component', () => {
            const wrapper = mount(
                <QuickInput
                    value='mock'
                    clearable={true}
                    clearableTooltipText={
                        <span>{'Custom'}</span>
                    }
                    onClear={() => {}}
                />,
            );

            expect(wrapper.find('.input-clear')).toMatchSnapshot();
        });
    });

    test('should dismiss clear button', () => {
        const focusFn = jest.fn();
        class MockComp extends React.PureComponent {
            focus = focusFn;
            render() {
                return <div/>;
            }
        }
        const wrapper = mount(
            <QuickInput
                value='mock'
                clearable={true}
                onClear={() => {}}
                inputComponent={MockComp}
            />,
        );

        wrapper.setProps({onClear: () => wrapper.setProps({value: ''})});
        expect(wrapper.find('.input-clear').exists()).toBe(true);

        wrapper.find('.input-clear').simulate('click');
        expect(wrapper.find('.input-clear').exists()).toBe(false);
        expect(focusFn).toBeCalled();
    });
});
