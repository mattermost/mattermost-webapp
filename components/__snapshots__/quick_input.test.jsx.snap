// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/QuickInput should render clear button with customized tooltip component 1`] = `
<div
  className="input-clear visible"
  onClick={[Function]}
>
  <OverlayTrigger
    defaultOverlayShown={false}
    delayShow={400}
    overlay={
      <Tooltip
        bsClass="tooltip"
        id="InputClearTooltip"
        placement="right"
      >
        <span>
          Custom
        </span>
      </Tooltip>
    }
    placement="bottom"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      delayShow={400}
      overlay={
        <OverlayWrapper
          bsClass="tooltip"
          id="InputClearTooltip"
          intl={null}
          placement="right"
        >
          <span>
            Custom
          </span>
        </OverlayWrapper>
      }
      placement="bottom"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <span
        aria-hidden="true"
        className="input-clear-x"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        ×
      </span>
    </OverlayTrigger>
  </OverlayTrigger>
</div>
`;

exports[`components/QuickInput should render clear button with customized tooltip text 1`] = `
<div
  className="input-clear visible"
  onClick={[Function]}
>
  <OverlayTrigger
    defaultOverlayShown={false}
    delayShow={400}
    overlay={
      <Tooltip
        bsClass="tooltip"
        id="InputClearTooltip"
        placement="right"
      >
        Custom
      </Tooltip>
    }
    placement="bottom"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      delayShow={400}
      overlay={
        <OverlayWrapper
          bsClass="tooltip"
          id="InputClearTooltip"
          intl={null}
          placement="right"
        >
          Custom
        </OverlayWrapper>
      }
      placement="bottom"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <span
        aria-hidden="true"
        className="input-clear-x"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        ×
      </span>
    </OverlayTrigger>
  </OverlayTrigger>
</div>
`;

exports[`components/QuickInput should render clear button with default tooltip text 1`] = `
<div
  className="input-clear visible"
  onClick={[Function]}
>
  <OverlayTrigger
    defaultOverlayShown={false}
    delayShow={400}
    overlay={
      <Tooltip
        bsClass="tooltip"
        id="InputClearTooltip"
        placement="right"
      >
        <FormattedMessage
          defaultMessage="Clear"
          id="input.clear"
          values={Object {}}
        />
      </Tooltip>
    }
    placement="bottom"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      delayShow={400}
      overlay={
        <OverlayWrapper
          bsClass="tooltip"
          id="InputClearTooltip"
          intl={null}
          placement="right"
        >
          <FormattedMessage
            defaultMessage="Clear"
            id="input.clear"
            values={Object {}}
          />
        </OverlayWrapper>
      }
      placement="bottom"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <span
        aria-hidden="true"
        className="input-clear-x"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        ×
      </span>
    </OverlayTrigger>
  </OverlayTrigger>
</div>
`;
